tests/functions.bash

```diff
@@ -213,15 +215,61 @@ test_language()
     # Get the new number of issues
     output=$(curl -su "admin:$SONARQUBE_ADMIN_PASSWORD" \
                     "$SONARQUBE_LOCAL_URL/api/issues/search?componentKeys=$projectKey")
-    issues=$(echo -e "$output" | jq '.issues | map(select(.status == "OPEN")) | length')
+    issues=$(echo -e "$output" | jq '.issues | map(select(.status == "OPEN"  or .status == "TO_REVIEW")) | length')
     if [ "$issues" -ne "$nbIssuesCnesQp" ]
     then
-        log "$ERROR" "Failed: there should be $nbIssuesCnesQp issues on the $languageName dummy project with the $cnesQp QP."
+        log "$ERROR" "Failed: there should be $nbIssuesCnesQp issues on the $languageName dummy project with the $cnesQp QP but $issues were found"
         log "$ERROR" "curl -su admin:$SONARQUBE_ADMIN_PASSWORD $SONARQUBE_LOCAL_URL/api/issues/search?componentKeys=$projectKey"
-        >&2 echo -e "$output"
+        echo -e "$output" | >&2 jq
         return 1
     fi
 
+    log "$INFO" "Analysis with $cnesQp QP ran as expected."
     log "$INFO" "Analyses succeeded, $languageName is supported."
     return 0
 }
+
+# test_analysis_tool
+#
+# This function tests that the image can run a
+# specified code analyzer and that it keeps producing
+# the same result given the same source code.
+#
+# Parameters:
+#   1: tool name
+#   2: tool command line
+#   3: analysis results reference file
+#   4: temporary results file
+#
+# Example:
+#   $ cmd="pylint -f json --rcfile=/opt/python/pylintrc_RNC_sonar_2017_A_B tests//src/*.py"
+#   $ test_analysis_tool "pylint" "$cmd" "tests/python/reference-pylint-results.json" "tests/python/tmp-pylint-results.json"
+test_analysis_tool()
+{
+    # Args
+    tool=$1
+    cmd=$2
+    ref_file=$3
+    tmp_file=$4
+
+    # Run an analysis with the tool
+    docker run --rm -u "$(id -u):$(id -g)" \
+                -v "$(pwd):/usr/src" \
+                lequal/sonar-scanner \
+                $cmd \
```

Comment:
Codacy found an issue: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/gh/cnescatlab/sonar-scanner/pullRequest?prid=6037747)